fw/tls.c

```diff
@@ -334,19 +331,25 @@ tfw_tls_encrypt(struct sock *sk, struct sk_buff *skb, unsigned int mss_now,
 	WARN_ON_ONCE(next == skb);
 	if (skb_tail == skb) {
 		r = ss_skb_expand_head_tail(skb->next, skb, head_sz, TTLS_TAG_LEN);
-		if (r < 0)
+		if (r < 0) {
+			tcb->end_seq -= head_sz;
```

Comment:
Just question, maybe we can do `tcb->end_seq += head_sz` after success work with skb?

Reply:
It seems we need correct `tcb->end_seq` for `tfw_tcp_propagate_dseq()`.

Reply:
Yes it is necessary for tfw_tcp_propagate_dseq